pipenv/patched/piptools/repositories/pypi.py

```diff
@@ -167,8 +167,18 @@ def find_best_match(self, ireq, prereleases=None):
         py_version = parse_version(os.environ.get('PIP_PYTHON_VERSION', str(sys.version_info[:3])))
         all_candidates = []
         for c in self.find_all_candidates(ireq.name):
-            if c.requires_python and not SpecifierSet(c.requires_python).contains(py_version):
-                continue
+            if c.requires_python:
+                # Old specifications had people setting this to single digits
+                # which is effectively the same as '>=digit,<digit+1'
+                if len(c.requires_python) == 1 and c.requires_python.isdigit():
```

Comment:
The `len(c.requires_python) ==1` part looks redundant to me, just `isdigit()` is enough. It does not matter practically, but if Python ever get to 10.0 we probably want to support that as well.